### DISCUSSION TITLE
VID:PID for Maiana USB adapter

### DISCUSSION THREAD

**Discussion by mgrouch** [peterantypas/maiana#97] [Q&A]
I’d like to add udev rule for Maiana AIS USB adapter into BBN OS so it would recognize it to make it plug and play. 

Can someone post output of 

lsusb 

when MAIANA is connected to usb on Linux?

looking for productId, vendorId and product string. 

Thanks,
—MG

**Comment by gah67** [comment]
If no-one else posts it I will do it later this week when I go down to my boat

**Comment by peterantypas** [comment]
I'm afraid it won't be much help. It's a generic WCH340 USB to serial adapter:

Bus 003 Device 037: ID 1a86:7523 QinHeng Electronics HL-340 USB-Serial adapter

**Comment by mgrouch** [comment]
Then it can't be made plug in play. 
I have the same line in lsusb output and it's used by totally different device.

**Comment by gah67** [comment]
thinking out loud here ... could you have a udev rule that triggers a script that enumerates the generic serial adaptors and identifies them based on a command/response and then renames the port to something useful

**Comment by mgrouch** [comment]
Kernel assigns names to the devices,
udev rule can launch a script to create a more human readable alias as symbolic link. Probing the device by writing into it might have undesirable effects with devices. I think it’s better to stay away from it.
